src/utils/common.ts

```diff
@@ -242,21 +242,23 @@ export type TransactionCostEstimate = {
  * @param provider A valid ethers provider - will be used to reason the gas price.
  * @param gasMarkup Markup on the estimated gas cost. For example, 0.2 will increase this resulting value 1.2x.
  * @param gasPrice A manually provided gas price - if set, this function will not resolve the current gas price.
+ * @param chainId The chain ID of the network that the transaction will be submitted to.
  * @returns Estimated cost in units of gas and the underlying gas token (gasPrice * estimatedGasUnits).
  */
 export async function estimateTotalGasRequiredByUnsignedTransaction(
   unsignedTx: PopulatedTransaction,
   senderAddress: string,
   provider: providers.Provider | L2Provider<providers.Provider>,
   gasMarkup: number,
-  gasPrice?: BigNumberish
+  gasPrice?: BigNumberish,
+  chainId?: number
 ): Promise<TransactionCostEstimate> {
   assert(
     gasMarkup > -1 && gasMarkup <= 4,
     `Require -1.0 < Gas Markup (${gasMarkup}) <= 4.0 for a total gas multiplier within (0, +5.0]`
   );
   const gasTotalMultiplier = toBNWei(1.0 + gasMarkup);
-  const { chainId } = await provider.getNetwork();
+  chainId ??= (await provider.getNetwork()).chainId;
```

Comment:
Did you confirm whether this actually produced an RPC call? I'd understood that we are using StaticJsonRpcProvider instances under the hood, and they _should_ suppress chainId lookups because they cache it internally. But it's also harder to know when the provider is typed broadly as `Provider` - and this means we don't get a concrete guarantee on it.

Reply:
No, you are right actually. I checked and there is no `eth_chainId` call. I will revert this change then

![image](https://github.com/across-protocol/sdk/assets/17645687/f765e4da-27c1-473f-b51f-4c82cdd42a12)